### PR TITLE
add enumeration for flags

### PR DESCRIPTION
**Pull request by hyunSo**
Created an enumeration for flags.
Please check for tiny change @cvanderw @dreamchaser3 !

### CODE REVIEW

File: NodejsConsoleApp1/app.js [hunk 1]
```diff
@@ -21,21 +21,25 @@ var sqlConnection = mysql.createConnection({
...
+    REQUEST_GAME_JOIN_USER_LIST_FLAG: 4,
+    REQUEST_GAME_TREASURE_INFO_FLAG: 5,
+    REQUEST_USER_INFO_FLAG: 6,
+
+    SET_USER_INFO_FLAG: 11,
+    SET_GAME_INFO_FLAG: 12,
+    SET_GAME_STATUS_END_FLAG: 13,
+    SET_USER_JOIN_FLAG: 14,
+    SET_USER_GET_TREASURE_FLAG: 15,
+};
```
Comment (dreamchaser3) [inline NodejsConsoleApp1/app.js]: @hyunSo oh..it's beautiful

**Comment by dreamchaser3** [comment]
@hyunSo It looks better. Thanks!! Beautiful! Wonderful!ㅋㅋ

File: NodejsConsoleApp1/app.js [hunk 2]
```diff
@@ -21,21 +21,25 @@ var sqlConnection = mysql.createConnection({
...
-* const	SET_GAME_STATUS_END_FLAG	13
-* const	SET_USER_JOIN_FLAG	14
-* const	SET_USER_GET_TREASURE_FLAG	15
-*/
+/**
+ * An enum for communication protocol flags
+ * @enum {number}
+ */
+const Flag = {
+    REQUEST_USER_TREASURE_LIST_FLAG: 1,
```
Comment (cvanderw) [inline NodejsConsoleApp1/app.js]: Suggestion: Remove "_FLAG" from all flags. When you use the flag you have to reference the containing Flag name anyways so showing Flag twice is redundant.

This:
`Flag.REQUEST_USER_TREASURE_LIST`

Looks preferable to:
`Flag.REQUEST_USER_TREASURE_LIST_FLAG`

File: NodejsConsoleApp1/app.js [hunk 3]
```diff
@@ -57,20 +61,19 @@ wsServer.on('request', function (request) {
 
                 // Server >> Clients according to flag
                 switch (flag) {
-
                     // TODO: Write all case process like 'case 6' below.
 
-                    case 6:// Get the user's information and send to Client.
+                    case Flag.REQUEST_USER_INFO_FLAG:// Get the user's information and send to Client.
```
Comment (cvanderw) [inline NodejsConsoleApp1/app.js]: Remove the comment. It's not necessary anymore now that the flag name is clearer.

File: NodejsConsoleApp1/app.js [hunk 4]
```diff
@@ -21,21 +21,25 @@ var sqlConnection = mysql.createConnection({
...
-* const	REQUEST_USER_INFO_FLAG	6
 
-* const	SET_USER_INFO_FLAG	11
-* const	SET_GAME_INFO_FLAG	12
-* const	SET_GAME_STATUS_END_FLAG	13
-* const	SET_USER_JOIN_FLAG	14
-* const	SET_USER_GET_TREASURE_FLAG	15
-*/
+/**
+ * An enum for communication protocol flags
```
Comment (cvanderw) [inline NodejsConsoleApp1/app.js]: End sentence in period.

**Review by cvanderw — COMMENTED** [review]
Looks great! :star:

A few minor comments.

**Review by cvanderw — APPROVED** [review]
Great idea to add a .gitignore file!

**Comment by hyunSo** [comment]
Lovely .gitignore !
I changed comments and will be waiting for merge @dreamchaser3 @cvanderw !

**Review by cvanderw — APPROVED** [review]
Looks great! 🌟 Approved!

I have a recommendation for future commit changes. Please include slightly more descriptive comments. Currently your comments are of the form "change comment issues". I think "fix flag names and comments" would be a bit more descriptive in this case.

Of course, if you have to make many other changes then it doesn't make sense to list all the changes out in the commit message. But in general I would recommend being more descriptive. That helps the reviewers know precisely what to expect in the commit.

**Pull request merged**